### PR TITLE
Add improved cookie API release notes

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

Updates v0.47 release notes for the browser cookie API.

## Why?

We improved the cookie API by adding a new `context.cookies([urls])` method. Introduced a `Cookie` class. And fixed a few bugs.

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`make ci-like-lint`) and all checks pass.
- [ ] I have run tests locally (`make tests`) and all tests pass.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

- https://github.com/grafana/xk6-browser/issues/6
- https://github.com/grafana/k6-docs/pull/1328
- https://github.com/grafana/k6-DefinitelyTyped/pull/14

### CODE REVIEW

File: release notes/v0.47.0.md [hunk 1]
```diff
@@ -37,6 +37,16 @@ Thanks, @sapphire-janrain, for contribution!
 
 In some workflows the reflection call should also include some metadata. This PR adds [a new connection parameter `reflectMetadata`](https://k6.io/docs/javascript-api/k6-net-grpc/client/client-connect/#connectparams) that allows to specify the metadata to be sent with the reflection call.
 
+### Improved the browser module's cookie API
```
Comment (ankur22) [inline release notes/v0.47.0.md:40]: Should the list of PRs be added to the end of this? Then we can shorten this to something like

```suggestion
Improve the cookie API
```

This suggestion is going by what i see elsewhere in the release notes.

File: release notes/v0.47.0.md [hunk 2]
```diff
@@ -37,6 +37,16 @@ Thanks, @sapphire-janrain, for contribution!
...
+
+We added a new [`browser.cookies([urls])`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookies/) method that returns all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). The new API also supports filtering cookies by URL. We also fixed a bug where the `expires` field was not being set correctly while adding cookies using the [`context.addCookie()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/addcookies/) method. Lastly, we fixed the existing [`context.clearCookies()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/clearcookies) method to clear all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). With all these changes, the new API is now more consistent with the [Playwright API](https://playwright.dev/docs/api/class-browsercontext).
```
Comment (ankur22) [inline release notes/v0.47.0.md:44]: Should we add an example of how to work with the new feature (`browser.cookies([urls])`)?

File: release notes/v0.47.0.md [hunk 3]
```diff
@@ -37,6 +37,16 @@ Thanks, @sapphire-janrain, for contribution!
 
 In some workflows the reflection call should also include some metadata. This PR adds [a new connection parameter `reflectMetadata`](https://k6.io/docs/javascript-api/k6-net-grpc/client/client-connect/#connectparams) that allows to specify the metadata to be sent with the reflection call.
 
+### Improved the browser module's cookie API
+
+The browser module now provides a more complete and robust API for handling cookies. We stabilized the cookie API by defining a new [`Cookie` class](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookie) that can be used while creating and retrieving cookies.
```
Comment (ankur22) [inline release notes/v0.47.0.md:42]: Should the new feature go under the new feature section, and the rest of the changes go in the improvements section? In one line a reference to the improvements can be mentioned in the new section so readers know that there were improvements made to existing APIs.

File: release notes/v0.47.0.md [hunk 4]
```diff
@@ -50,6 +60,9 @@ _Format as `<number> <present_verb> <object>. <credit>`_:
 
 - [xk6-grpc#47](https://github.com/grafana/xk6-grpc/pull/47) fix the premature closing of the stream when a stream's client has finished sending. Thanks, @thiagodpf, for reporting!
 - [#3344](https://github.com/grafana/k6/pull/3344) and [xk6-grpc#49](https://github.com/grafana/xk6-grpc/pull/49) adds (fixes) the support of google protobuf wrappers. Thanks, @zibul444, for reporting!
+- [browser#1040](https://github.com/grafana/xk6-browser/pull/1040) fix the `context.clearCookies()` method to clear all cookies from the current browser context.
+- [browser#1031](https://github.com/grafana/xk6-browser/pull/1031) fix the `expires` field while adding cookies using the `context.addCookie()` method.
+- [browser#1039](https://github.com/grafana/xk6-browser/pull/1039) fix Goja conversions while adding and retriving cookies.
```
Comment (ankur22) [inline release notes/v0.47.0.md:65]: This looks good to me 👍

**Review by ankur22 — COMMENTED** [review]
Thanks for the cookie work to the release notes! I've left some suggestions. I'm not 100% sure on them, and it does feel quite subjective.

File: release notes/v0.47.0.md (hunk 1 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:40]: As far as I can see, a big new feature is written as a heading; like we did with Web Vitals [before](https://github.com/grafana/k6/blob/39f0e6882228a061dbc495344dda5cc1261e7942/release%20notes/v0.44.0.md). Or, do you mean something else? 🤔

File: release notes/v0.47.0.md (hunk 2 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:44]: We link to the documentation. I can add an example here if you think it makes sense.

File: release notes/v0.47.0.md (hunk 3 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:42]: That makes sense. But, I want to give a comprehensive explanation of what we improved. Like we did [before here](https://github.com/grafana/k6/blob/39f0e6882228a061dbc495344dda5cc1261e7942/release%20notes/v0.46.0.md).

File: release notes/v0.47.0.md (hunk 1 above)
Comment (ankur22) [reply release notes/v0.47.0.md:40]: I'm not 100% sure on shortening the heading tbh.

What about adding the list of issues after the heading, which seems to be how other new features are added.

```suggestion
### Improved the browser module's cookie API browser#1, browser#2...
```

File: release notes/v0.47.0.md (hunk 2 above)
Comment (ankur22) [reply release notes/v0.47.0.md:44]: Just going by what we've done in the past, it seems that generally we add a short code snippet like what we did for `addCookies` in this [release note](https://github.com/grafana/k6/releases/tag/v0.44.0).

File: release notes/v0.47.0.md (hunk 1 above)
Comment (ankur22) [reply release notes/v0.47.0.md:40]: Fair enough. Last suggestion on this 🙂 What about adding the issue links closer to when they're first mentioned e.g.:

```suggestion
The browser module now provides a more complete and robust API for handling cookies. We stabilized the cookie API by defining a new [`Cookie` class](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookie) (browser#1, browser#2)
```

and

```suggestion
We added a new [`browser.cookies([urls])`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookies/) (browser#3) method that returns all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). The new API also supports filtering cookies by URL (browser#4).
```

Then we wouldn't need the related link block at the end of this section, which doesn't feel like it fits with how we generally incorporate the pr/issue links.

File: release notes/v0.47.0.md (hunk 2 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:44]: WDYT about this example? [a2076be](https://github.com/grafana/k6/pull/3352/commits/a2076be60b1d73aabfdc0145a7c8c56ebb1ca2d5)

Comment (ankur22) [reply release notes/v0.47.0.md:44]: Thanks!

File: release notes/v0.47.0.md (hunk 1 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:40]: Can you write it as a suggestion because I'm not sure I understand 😮‍💨

File: release notes/v0.47.0.md [hunk 5]
```diff
@@ -37,6 +37,32 @@ Thanks, @sapphire-janrain, for contribution!
...
+
+We added a new [`browser.cookies([urls])`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookies/) method that returns all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). The new API also supports filtering cookies by URL. We also fixed a bug where the `expires` field was not being set correctly while adding cookies using the [`context.addCookie()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/addcookies/) method. Lastly, we fixed the existing [`context.clearCookies()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/clearcookies) method to clear all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). With all these changes, the new API is now more consistent with the [Playwright API](https://playwright.dev/docs/api/class-browsercontext).
```
Comment (ankur22) [inline release notes/v0.47.0.md:44]: This suggestion is linked to https://github.com/grafana/k6/pull/3352/files#r1337145487

```suggestion
We added a new [`browserContext.cookies([urls])`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookies/) ([browser#1005](https://github.com/grafana/xk6-browser/pull/1005)) method that returns all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). The new API also supports filtering cookies by URL ([browser#1016](https://github.com/grafana/xk6-browser/pull/1016)). We also fixed a bug where the `expires` field was not being set correctly while adding cookies using the [`context.addCookie()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/addcookies/) method. Lastly, we fixed the existing [`context.clearCookies()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/clearcookies) method to clear all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). With all these changes, the new API is now more consistent with the [Playwright API](https://playwright.dev/docs/api/class-browsercontext).
```

Note that the PRs for `browser.cookies` and the filtering option are inlined. It's not super clear unfortunately from the suggestion :/

File: release notes/v0.47.0.md [hunk 6]
```diff
@@ -37,6 +37,32 @@ Thanks, @sapphire-janrain, for contribution!
 
 In some workflows the reflection call should also include some metadata. This PR adds [a new connection parameter `reflectMetadata`](https://k6.io/docs/javascript-api/k6-net-grpc/client/client-connect/#connectparams) that allows to specify the metadata to be sent with the reflection call.
 
+### Improved the browser module's cookie API
+
+The browser module now provides a more complete and robust API for handling cookies. We stabilized the cookie API by defining a new [`Cookie` class](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookie) that can be used while creating and retrieving cookies.
```
Comment (ankur22) [inline release notes/v0.47.0.md:42]: This suggestion is linked to https://github.com/grafana/k6/pull/3352/files#r1337145487

```suggestion
The browser module now provides a more complete and robust API for handling cookies. We stabilized the cookie API by defining a new [`Cookie` class](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookie) ([browser#1008](https://github.com/grafana/xk6-browser/pull/1008), [browser#1030](https://github.com/grafana/xk6-browser/pull/1030)) that can be used while creating and retrieving cookies.
```

File: release notes/v0.47.0.md (hunk 1 above)
Comment (ankur22) [reply release notes/v0.47.0.md:40]: See these:

- https://github.com/grafana/k6/pull/3352/files#r1337163681
- https://github.com/grafana/k6/pull/3352/files#r1337163639

File: release notes/v0.47.0.md (hunk 6 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:42]: Ah, I see now. You want to make the related PRs clearer 👍

File: release notes/v0.47.0.md (hunk 5 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:44]: Thanks much 🙇

File: release notes/v0.47.0.md (hunk 1 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:40]: Thanks! Check it out: [38c57bc](https://github.com/grafana/k6/pull/3352/commits/38c57bce967386f000b8b937496fb3985d47f470)

**Review by ankur22 — APPROVED** [review]
LGTM 🙇

File: release notes/v0.47.0.md [hunk 7]
```diff
@@ -37,6 +37,28 @@ Thanks, @sapphire-janrain, for contribution!
...
+
+We added a new [`browserContext.cookies([urls])`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookies/) ([browser#1005](https://github.com/grafana/xk6-browser/pull/1005)) method that returns all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). The new API also supports filtering cookies by URL ([browser#1016](https://github.com/grafana/xk6-browser/pull/1016)). We also fixed a bug where the `expires` field was not being set correctly while adding cookies using the [`context.addCookie()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/addcookies/) method. Lastly, we fixed the existing [`context.clearCookies()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/clearcookies) method to clear all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). With all these changes, the new API is now more consistent with the [Playwright API](https://playwright.dev/docs/api/class-browsercontext).
```
Comment (ka3de) [inline release notes/v0.47.0.md:44]: I don't particularly like to mention Playwright's API here. But no strong opinion.
```suggestion
We added a new [`browserContext.cookies([urls])`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/cookies/) ([browser#1005](https://github.com/grafana/xk6-browser/pull/1005)) method that returns all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext). The new API also supports filtering cookies by URL ([browser#1016](https://github.com/grafana/xk6-browser/pull/1016)). We also fixed a bug where the `expires` field was not being set correctly while adding cookies using the [`context.addCookie()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/addcookies/) method. Lastly, we fixed the existing [`context.clearCookies()`](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext/clearcookies) method to clear all cookies from the current [browser context](https://k6.io/docs/javascript-api/k6-experimental/browser/browsercontext).
```

**Review by ka3de — APPROVED** [review]
LGTM.

File: release notes/v0.47.0.md (hunk 7 above)
Comment (inancgumus) [reply release notes/v0.47.0.md:44]: We offer _rough_ compatibility with Playwright. Could you explain why you don't like it?
